mlflow/tracking/fluent.py

```diff
@@ -278,15 +279,15 @@ def start_run(
 
         exp_id_for_run = experiment_id if experiment_id is not None else _get_experiment_id()
 
-        user_specified_tags = tags or {}
+        user_specified_tags = deepcopy(tags) or {}
```

Comment:
do we need deepcopy ? can we use tags.copy() ?

Reply:
@WeichenXu123 I think `deepcopy` is safer because `tags` can be a nested dictionary.